src/sql/src/plan/statement/dml.rs

```diff
@@ -258,7 +258,7 @@ pub fn describe_explain_plan(
 ) -> Result<StatementDesc, PlanError> {
     let mut relation_desc = RelationDesc::empty();
 
-    match stage {
+    match stage.unwrap_or(ExplainStage::GlobalPlan) {
```

Comment:
This default appears in two places and I don't like that. WDYT about having all of these defaults be `impl` fns in `statement.rs`? The downside is that the parser crate will have to know a bit about planning (bad), but there'd only be a single place where they are defined (good). I'd really like those impl blocks to be in, say, this file, but I'm not aware of a non-awkward way to do that.

The awkward way would be `fn explain_stage_default(stmt: &ExplainStatement)` or something, and having a ton of those littered all over here seems real bad. There's prior art to doing this, but I'm not sure we want to follow it:
https://github.com/MaterializeInc/materialize/blob/1104524b3e17a93d9228a984072a7c549afeab31/src/sql-parser/src/ast/defs/statement.rs#L4540

Reply:
Totally agree, def want to define the defaults in one place. I think the approach of having the defaults be `impl` fns in `statement.rs` works! An alternative I'll propose is what if planning defined its own trait, e.g.

```
/// Default value used when planning a statement, be careful about changing this!
pub trait PlanningDefault {
    fn plan_default() -> Self;
}
```

This should allow us to define the default in the `sql` crate, and doesn't seem too awkward?

Taking it one step further we could add an `*Ext` trait to provide an `unwrap_or_plan_default()` method on `Option`s, e.g.

```
pub trait PlanningOptionExt<T> {
    fn unwrap_or_plan_default(self) -> T
        where T: PlanningDefault;
}

impl<T> PlanningOptionExt<T> for Option<T> {
    fn unwrap_or_plan_default(self) -> T
        where T: PlanningDefault
    {
        self.unwrap_or_else(|| T::plan_default())
    }
}
```

I'll leave this up to you though, I think putting some values in `statement.rs` would work just fine :)

Reply:
Thanks! Made impl blocks for these.